client/rest/src/main/java/org/elasticsearch/client/ResponseException.java

```diff
@@ -33,7 +33,7 @@
 
     private Response response;
 
-    ResponseException(Response response) throws IOException {
+    public ResponseException(Response response) throws IOException {
```

Comment:
why make response non final with a package private constructor and leave ResponseException final but add a public constructor to it? Should we do the same with both?